gpt4all-bindings/python/docs/gpt4all_cli.md

```diff
@@ -0,0 +1,198 @@
+```
+
+##### Linux
+
+On Linux, a Python installation is often split into several packages and not all are necessarily
+installed by default. For example, on Debian/Ubuntu and derived distros, you will want to ensure
+their presence with the following:
+```shell
+sudo apt-get install python3-venv python3-pip
+```
+The next steps are similar to the other platforms:
+```shell
+python3 -m venv gpt4all-cli
+. gpt4all-cli/bin/activate
+python3 -m pip install gpt4all typer
+```
+On other distros, the situation might be different. Especially the package names can vary a lot.
+You'll have to look it up in the documentation, software directory, or package search.
+
+#### User Installation
+##### macOS
+
+There are at least three ways to have a Python installation on _macOS_, and possibly not all of them
+provide a full installation of Python and its tools. When in doubt, try the following:
+```shell
+python3 -m pip --help
+```
+That should print the help for the `pip` command. If it doesn't, consult the documentation of your
+Python installation on how to enable them, or download a separate Python variant, for example try an
+[unified installer package from python.org][python.org-downloads].
+
+Once ready, do:
+```shell
+python3 -m pip install --user --upgrade gpt4all typer
+```
+
+##### Windows
+
+Download the [official installer from python.org][python.org-downloads] if Python isn't already
+present on your system. It includes all the necessary components. Run:
+```shell
+py -3 -m pip install --user --upgrade gpt4all typer
+```
+
+##### Linux
+
+On Linux, a Python installation is often split into several packages and not all are necessarily
+installed by default. For example, on Debian/Ubuntu and derived distros, you will want to ensure
+their presence with the following:
+```shell
+sudo apt-get install python3-pip
+```
+The next steps are similar to the other platforms:
+```shell
+python3 -m pip install --user --upgrade gpt4all typer
+```
+On other distros, the situation might be different. Especially the package names can vary a lot.
+You'll have to look it up in the documentation, software directory, or package search.
+
+## Running the CLI
+
+The CLI is a self-contained script called [app.py]. As such, you can [download][app.py-download]
+and save it anywhere you like, as long as the Python interpreter has access to the mentioned
+dependencies.
+
+Note: different platforms can have slightly different ways to start Python. Whereas below the
+interpreter command is written as `python` you typically want to type instead:
+- On _Unix-like_ systems: `python3`
+- On _Windows_: `py -3`
+
+The simplest way to start the CLI is:
+```shell
+python app.py repl
+```
+This automatically selects the [groovy] model and downloads it into the `.cache/gpt4all/` folder
+of your home directory, if not already present.
+
+[groovy]: https://github.com/nomic-ai/gpt4all/blob/2b6cc99a31a124f1f27f2dc6515b94b84d35b254/gpt4all-chat/metadata/models.json#L3-L8

```

Comment:
might make sense to link the the HF [model card](https://huggingface.co/nomic-ai/gpt4all-j) instead of the models.json? (the latest version of that hf hub repo is groovy, but it also the old jazzy/breezy gpt-j models in branches)

Reply:
Yes, that's probably more fitting.